integrationTests/mock/contracts/elrond.go

```diff
@@ -18,33 +20,40 @@ type ElrondContract struct {
 	*mock.Contract
 	mutWhitelisted       sync.RWMutex
 	whitelistedAddresses map[string]struct{}
+	tokensHandler        TokensHandler
 }
 
 // NewElrondContract defines the mocked Elrond contract functions
-func NewElrondContract(address string) *ElrondContract {
+func NewElrondContract(address string, th TokensHandler) (*ElrondContract, error) {
+	if check.IfNil(th) {
+		return nil, ErrNilTokenHandler
+	}
+
 	ec := &ElrondContract{
 		Contract:             mock.NewContract(address),
 		whitelistedAddresses: make(map[string]struct{}),
+		tokensHandler:        th,
 	}
 
 	ec.createContractFunctions()
 
-	return ec
+	return ec, nil
 }
 
 func (ec *ElrondContract) createContractFunctions() {
 	ec.AddHandler("getCurrentTxBatch", ec.getCurrentTxBatch)
 	ec.AddHandler("userRole", ec.userRole)
+	ec.AddHandler("getTokenIdForErc20Address", ec.getTokenIdForErc20Address)
 }
 
 func (ec *ElrondContract) getCurrentTxBatch(caller string, value string, arguments ...string) ([][]byte, error) {
-	log.Warn("getCurrentTxBatch", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+	log.Debug("getCurrentTxBatch", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
```

Comment:
I think the logger will automatically do a `fmt.Sprintf("%v")` over `arguments`

Reply:
done